### PR TITLE
Add a grid movement example

### PR DESCRIPTION
**Pull request by Masonstiller**
Adds example of basic grid movement

### CODE REVIEW

File: examples/grid.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import pygame
```
Comment (novialriptide) [inline examples/grid.py:1]: In other example files, they import `pygame` as `pg`, I don't like it, but it's better to stay consistent.

File: examples/grid.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def __init__(self):
+        pygame.init()
+        self.clock = pygame.time.Clock()
+        pygame.display.set_caption(TITLE)
+        self.surface = pygame.display.set_mode((WINDOW_WIDTH, WINDOW_HEIGHT))
+        self.loop = True
+        self.player = Player(self.surface)
+
+    def main(self):
+        while self.loop:
```
Comment (novialriptide) [inline examples/grid.py:36]: I think it's better to merge the two functions `events()` and `draw()` together.

File: examples/grid.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    def draw(self):
+        self.surface.fill((0, 0, 0))
+        for row in range(TILES_HORIZONTAL):
+            for col in range(row % 2, TILES_HORIZONTAL, 2):
+                pygame.draw.rect(self.surface, (40, 40, 40), (row * TILE_SIZE, col * TILE_SIZE, TILE_SIZE, TILE_SIZE))
+        self.player.draw()
+        pygame.display.update()
+
+
+if __name__ == "__main__":
```
Comment (novialriptide) [inline examples/grid.py:60]: A question to future reviewers, should we keep this in? `if __name__ == "__main__":` isn't in other examples.

**Review by novialriptide — CHANGES_REQUESTED** [review]
Great PR, I haven't run the example yet, but I have a few nitpicks here.

File: examples/grid.py (hunk 3 above)
Comment (ankith26) [reply examples/grid.py:60]: I think this is good practice... so maybe all examples should have it?

Comment (novialriptide) [reply examples/grid.py:60]: Good practice indeed, but `import pygame as pg` is a little wack as well imo.

**Comment by MyreMylar** [comment]
looks like this needs a pass through the formatting linters with a:

```
black examples/grid.py
```

Are you still interested in finishing this off @Masonstiller ?

**Comment by Masonstiller** [comment]
I’ll submit an improved version based off the feedback I’ve gotten in the next couple of days. Sorry for the long pause in development have been crazy busy lately.

File: examples/grid.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import pygame as pg
+
+TITLE = "Grid"
+TILES_HORIZONTAL = 10
+TILES_VERTICAL = 10
+TILE_SIZE = 100
+WINDOW_WIDTH = 1000
+WINDOW_HEIGHT = 1000
```
Comment (Starbuck5) [inline examples/grid.py:8]: This window height will be too big for some people (like if they have display scaling from the OS). Maybe something like 800x800 would be safer.

File: examples/grid.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    self.loop = False
+            elif event.type == pg.MOUSEBUTTONUP:
+                pos = pg.mouse.get_pos()
+                self.player.move(pos)
+        pg.display.update()
+
+
+if __name__ == "__main__":
+    mygame = Game()
+    mygame.main()
```
Comment (Starbuck5) [inline examples/grid.py:59]: Needs a trailing newline.

Comment (Starbuck5) [reply examples/grid.py:59]: Also should have a `pg.quit()` at the end

File: examples/grid.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    def __init__(self, surface):
+        self.surface = surface
+        self.pos = (50, 50)
+
+    def draw(self):
+        pg.draw.circle(self.surface, (255, 255, 255), self.pos, 50)
+
+    def move(self, target):
+        x = (100 * (target[0] // 100)) + 50
+        y = (100 * (target[1] // 100)) + 50
```
Comment (Starbuck5) [inline examples/grid.py:21]: Is this the same as a math.floor?

Comment (Masonstiller) [reply examples/grid.py]: Yes. Should I use math.floor for syntax reasons?

File: examples/grid.py (hunk 4 above)
Comment (Masonstiller) [reply examples/grid.py]: Good idea.

File: examples/grid.py (hunk 6 above)
Comment (novialriptide) [reply examples/grid.py]: @Masonstiller Yup

**Comment by MyreMylar** [comment]
This needs a:

```
black format examples/grid.py
```

to pass the linter CI up there, otherwise we can't merge it.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Just needs linting @Masonstiller.

**Review by illume — APPROVED** [review]
Thanks 👍

**Comment by illume** [comment]
Merged in https://github.com/pygame/pygame/pull/3754

**Pull request closed without merging**